Sources/NIOCore/AsyncSequences/NIOAsyncWriter.swift

```diff
@@ -17,6 +17,9 @@ import DequeModule
 import NIOConcurrencyHelpers
 import _NIODataStructures
 
+@usableFromInline
+let yieldIDCounter = ManagedAtomic<UInt64>(0)
```

Comment:
This would probably be nicer as a static, but this is otherwise reasonable.

Reply:
Agreed but wasn't sure where to put it. The obvious place is `NIOAsyncWriter` but that's generic so can't have a static let.

Reply:
I am not sure if I agree with this move. I understand that we want to reduce allocations but IMO the correct move forward is to replace `SwiftAtomics.ManagedAtomic` with `Synchronisation.Atomic` which should avoid the allocation as well since it is `~Copyable`. We ought to be able to do this conditionally when the compiler is Swift 6. Though availability might be tricky. If availability blocks us we can move this into a global. I would suggest renaming though to scope it better e.g. `let _asyncWriterYieldIDCounter` or something like that

Reply:
I agree re: using `Synchronization.Atomic` but as noted, the availability makes this difficult. I've renamed the global though.